plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse.go

```diff
@@ -0,0 +1,498 @@
+			s.wg.Done()
+		}
+
+	case "keepalive":
+		// Keepalive events are ignored for the moment
+		s.Log.Debug("Received keepalive event")
+
+	default:
+		// Received a yet unsupported event type
+		s.Log.Debug("Received unsupported event")
+	}
+}
+
+// Init is for setup, and validating config.
+func (s *ctrlXDataLayerSSE) Init() error {
+	if s.Timeout == 0 {
+		s.Timeout = defaultTimeout
+	}
+	if time.Duration(s.Timeout) < time.Duration(defaultTimeout) {
+		s.Log.Errorf("Setting timeout must be at least %v seconds", time.Duration(defaultTimeout).Seconds())
+		return errors.New("timeout configuration error")
+	}
+
+	for i, sub := range s.Subscription {
+		err := choice.Check(sub.QueueBehaviour, queueBehaviours)
+		if err != nil {
+			s.Log.Errorf("SSE subscription %v: setting 'queue_behaviour' %v is incorrectly configured", i, sub.QueueBehaviour)
+			s.Log.Infof("The right queue behaviour values are %v", queueBehaviours)
+			return err
+		}
+		err = choice.Check(sub.ValueChange, valueChanges)
+		if err != nil {
+			s.Log.Errorf("SSE subscription %v: setting 'value_change' %v is incorrectly configured", i, sub.ValueChange)
+			s.Log.Infof("The right value change values are %v", valueChanges)
+			return err
+		}
+	}
+
+	s.connection = &http.Client{
+		Timeout: time.Duration(s.Timeout),
+	}
+
+	return nil
+}
+
+// Start input as service, retain the accumulator, establish the connection
+// The reconnect will start if the connection is not established
+func (s *ctrlXDataLayerSSE) Start(acc telegraf.Accumulator) error {
+	s.connection.Transport = httpTransport(s.InsecureSkipVerify)
+
+	loginFailed := false
+	if !s.setAuthToken() {
+		s.Log.Errorf("hostname: %v, username: %v log in failed", s.Hostname, s.Username)
+		loginFailed = true
+	}
+
+	s.acc = acc
+
+	for k, l := range s.Subscription {
+		l.index = k
+
+		if !l.Enable {
+			continue
+		}
+
+		if len(l.Nodes) == 0 {
+			continue
+		}
+
+		if loginFailed {
+			s.reconnect(l)
+			continue
+		}
+
+		subURL := s.createSubscription(&l)
+		if subURL == "" {
+			s.reconnect(l)
+			continue
+		}
+
+		client := newClient(subURL, l, s.Log)
+		if s.autoReconnect {
+			client.onDisconnect(s.reconnect)
+		}
+		client.Headers["Authorization"] = s.token.String()
+		client.Connection.Transport = httpTransport(s.InsecureSkipVerify)
+		err := client.requestData(s, l.Measurement)
+		if err != nil {
+			s.reconnect(l)
+		}
+	}
+	return nil
+}
+
+// Stop input as service
+func (s *ctrlXDataLayerSSE) Stop() {
+	s.wg.Wait()
+	s.acc = nil
+}
+
+// Gather is called by telegraf to collect the metrics
+func (s *ctrlXDataLayerSSE) Gather(acc telegraf.Accumulator) error {
+	// Metrics are sent to the accumulator asynchronously in worker thread. So nothing to do here.
+	return nil
+}
+
+// init registers the plugin in telegraf
+func init() {
+	inputs.Add("ctrlx_datalayer_sse", func() telegraf.Input {
+		return &ctrlXDataLayerSSE{
+			autoReconnect: true,
+		}
+	})
```

Comment:
Init the default timeout here to allow overriding by the user
```suggestion
	inputs.Add("ctrlx_datalayer_sse", func() telegraf.Input {
		return &ctrlXDataLayerSSE{
			Timeout: defaultTimeout,
			autoReconnect: true,
		}
	})
```

Reply:
Done